src/middlewared/middlewared/alert/source/ctdb.py

```diff
@@ -19,3 +27,102 @@ class CtdbClusteredServiceAlertClass(AlertClass, SimpleOneShotAlertClass):
 
     async def delete(self, alerts, query):
         return []
+
+
+class ClusteredClockAlertClass(AlertClass):
+    category = AlertCategory.CLUSTERING
+    level = AlertLevel.WARNING
+    title = "Clustered time consistency check failed"
+    text = "%(errmsg)s"
+
+
+class ClusteredClockOffsetAlertSource(AlertSource):
+    schedule = IntervalSchedule(timedelta(hours=1))
+    run_on_backup_node = False
+
+    async def check(self):
+        def get_clock_realtime(entry):
+            return entry['clock_realtime']
+
+        def get_offset(entry):
+            return abs(entry['ntp_peer']['offset'])
+
+        if await self.middleware.call('smb.get_smb_ha_mode') != 'CLUSTERED':
```

Comment:
Use the new `cluster.utils.is_clustered` api endpoint instead? Not sure, up to you.